plugins/package-managers/swiftpm/src/main/kotlin/SwiftPmModel.kt

```diff
@@ -114,6 +132,35 @@ internal fun PinV1.toPackage(): Package {
     return createPackage(id, vcsInfo)
 }
 
+internal fun PinV2.toPackage(): Package {
+    val id = Identifier(
+        type = PACKAGE_TYPE,
+        namespace = "",
+        name = getCanonicalName(location),
+        version = state?.run {
+            when {
+                !version.isNullOrBlank() -> version
+                !revision.isNullOrBlank() -> "revision-$revision"
+                !branch.isNullOrBlank() -> "branch-$branch"
+                else -> ""
+            }
+        }.orEmpty()
+    )
+
+    val vcsInfoFromUrl = VcsHost.parseUrl(location)
+    val vcsInfo = if (vcsInfoFromUrl.revision.isBlank() && state != null) {
+        when {
+            !state.revision.isNullOrBlank() -> vcsInfoFromUrl.copy(revision = state.revision)
+            !state.version.isNullOrBlank() -> vcsInfoFromUrl.copy(revision = state.version)
+            else -> vcsInfoFromUrl
+        }
+    } else {
+        vcsInfoFromUrl
+    }
```

Comment:
Any plans to reduce the code duplication here with `PinV1.toPackage()`?

Reply:
Nothing concrete yet. I thought it could make sense to first look into version 3 format, and to see how the contract is for the properties of `state` and also considering `kind` property before using a common path.